NEWS

```diff
@@ -58,6 +58,13 @@ Changed Functionality
   ``frameworks/signatures/iso-9660`` which also increases the BOF buffer sufficiently.
   Note, doing so may increase memory and CPU usage significantly.
 
+- If a Spicy protocol analyzers feeds data into file analysis, it now
+  needs to call Zeek's `Files::register_protocol()` and provide a
+  callback for computing file handles. If that's missing, Zeek will
+  issue a warning. While this was not necessary in previous versions,
+  it aligns with the same requirement for traditional analyzers and
+  enables customizing file handles for protocol-specific semantics.
```

Comment:
Is there a way we could provide a sensible default in the template for protocol analyzers?

Reply:
I was wondering about that too. The thing is that this doesn't apply to most protocol analyzers, only those that feed data into file analysis. I guess we could leave some commented out code in the template?

Reply:
> I guess we could leave some commented out code in the template?

Yes, I believe that would be helpful, ideally with that comment explaining how `get_file_handle` would need to behave (maybe even expanding a bit of what is in the Zeek docs for e.g., `get_file_handle` which we should reference). I don't think we e.g., want to introduce code which counts files into the default template, and it seems hard to provide a reusable function.

Reply:
Ok, I'll work on the template.